5.5/docker-entrypoint.sh

```diff
@@ -1,13 +1,25 @@
 #!/bin/bash
 set -e
 
+get_option () {
+    local section=$1
+    local option=$2
+    local default=$3
+    ret=$(/usr/local/mysql/bin/my_print_defaults $section | grep '^--'${option}'=' | cut -d= -f2-)
+    [ -z $ret ] && ret=$default
+    echo $ret
+}
+
 if [ "${1:0:1}" = '-' ]; then
 	set -- mysqld "$@"
 fi
 
 if [ "$1" = 'mysqld' ]; then
-	# read DATADIR from the MySQL config
-	DATADIR="$("$@" --verbose --help 2>/dev/null | awk '$1 == "datadir" { print $2; exit }')"
+	# Get config
+        DATADIR="$("$@" --verbose --help 2>/dev/null | awk '$1 == "datadir" { print $2; exit }')"
+        SOCKET=$(get_option  mysqld socket "/tmp/mysql.sock")
+        HOSTNAME=$(hostname)
+        PIDFILE=$(get_option mysqld pid-file "$DATADIR/$HOSTNAME.pid")
```

Comment:
The pidfile option isn't included in the mysql config of the package used by this image.
We could append the option to the config file, but this seemed as clean a solution.